prow/gerrit/adapter/adapter.go

```diff
@@ -419,14 +420,41 @@ func (c *Controller) handleInRepoConfigError(err error, instance string, change
 }
 
 // processChange creates new presubmit/postsubmit prowjobs base off the gerrit changes
-func (c *Controller) processChange(logger logrus.FieldLogger, instance string, change client.ChangeInfo) error {
+func (c *Controller) processChange(logger logrus.FieldLogger, instance string, change client.ChangeInfo, lastUpdated time.Time) error {
 	tracker := time.Now()
 	defer func() {
 		// tracker will reset in `processChange` function, and this is the time
 		// taken for a gerrit change to be processed after the last reset.
 		logger.WithField("duration(s)", time.Since(tracker).Seconds()).Debug("Time taken for gerrit change to be processed.")
 	}()
 
+	if change.Revisions != nil {
+		var revisions []gerrit.RevisionInfo
+		for _, rev := range change.Revisions {
+			revisions = append(revisions, rev)
+		}
+		// sort revisions by create time descending
+		sort.Slice(revisions, func(i, j int) bool {
```

Comment:
I don't think you need to sort, as which one is evaluated earlier doesn't really make a difference. So the next for loop could just be:
```
var hasCodeChange bool
for _, rev := range revisions {
			if rev.Created.Time.After(lastUpdated) && rev.Kind != gerrit.NoCodeChange {
                                 hasCodeChange = true
				break
			}
		}
```

Reply:
Great simplifications, done.